docs/core-design.md

```diff
@@ -0,0 +1,354 @@
+
+
+| Reason name | Description |
+| ----- | ----- |
+| ConditionsNotReady | Other conditions on the object have “false” value. |
+| AdministrativelyDisabled | The PodNetwork’s Enabled field is set to false. |
+
+The provider’s implementation is responsible for the conditions life-cycle.
+
+##### Other Conditions
+
+We will not restrict adding other conditions to this object. This way implementations can further control the Readiness of PodNetwork by adding their own Condition and logic for it.
+
+##### Implementation consideration
+
+Implementations must enforce the Ready Condition from PodNetwork. It is up to the implementation for the solution, e.g. withdraw resource availability for the specific PodNetwork across Nodes.
+
+#### Provider
+
+The provider field gives the ability to uniquely identify what implementation (provider) is going to handle specific instances of PodNetwork objects. The value has to be in the form of an url. It is the implementer’s decision on how this field is going to be leveraged. They will decide how they behave when this field is empty and what specific value they are going to respect. This will dictate if a specific implementation can co-exist with other ones in the same cluster.
+
+#### Enabled
+
+The Enabled field is created to allow administrative control for a given PodNetwork (e.g. for migration). When set to false no new Pods can be attached to that PodNetwork.
+
+#### Parameters
+
+Parameters field is a free form configuration for the implementations to use and define how these fields are used.
+
+#### InUse state
+
+It is recommended to put PodNetwork objects that are referenced by at least one Pod in an InUse state. It is up to the implementation to do it and how that is implemented.
+
+#### Lifecycle
+
+A PodNetwork will be in following phases:
+
+1. Created/NotReady \- when PodNetwork is just created its Ready condition is false. Pods can reference such a PodNetwork, but will be in Pending state until the PodNetwork becomes Ready.  
+2. Ready \- when validation of the PodNetwork succeeded and Ready condition is set to true. Here Pods can start attaching to a PodNetwork.  
+3. InUse \- when there is a Pod that references a given PodNetwork.  
+4. Disabled \- when the user sets the Enabled field to false. Such PodNetwork should be NotReady, and no new Pods will be able to attach to such PodNetwork.
+
+### DRA integration
+
+This design defines DRA driver implementation standardization. This serves to allow unification on the usability of Multi-Network within Kubernetes Cluster and will allow for future features integrations.
+
+#### Device PodNetwork attribute
+
+This design standardizes for all drivers supporting PodNetwork to provide the “\<driver name\>/podNetwork” attribute for the corresponding device listed in DRA’s ResourceSlice object.
```

Comment:
if we want to make the attribute standard across drivers and use things like `matchAttributes` for scheduling we need to remove the driver suffix to make the attribute universal

```
networking.dra.x-k8s.io/podNetwork
```

or something like that